rotkehlchen/tasks/manager.py

```diff
@@ -538,6 +541,36 @@ def _maybe_update_snapshot_balances(self) -> Optional[list[gevent.Greenlet]]:
         """
         with self.database.conn.read_ctx() as cursor:
             if self.database.should_save_balances(cursor):
+                detect_tokens: dict[SupportedBlockchain, list[ChecksumEvmAddress]] = defaultdict(list)  # noqa: E501
+                if CachedSettings().get_settings().auto_detect_tokens is True:
+                    type_filter = [event_type.serialize() for event_type in [
+                        HistoryEventType.RECEIVE,
+                        HistoryEventType.WITHDRAWAL,
+                    ]]
+                    subtype_filter = [event_subtype.serialize() for event_subtype in [
+                        HistoryEventSubType.RECEIVE,
+                        HistoryEventSubType.REWARD,
+                        HistoryEventSubType.REMOVE_ASSET,
+                        HistoryEventSubType.MEV_REWARD,
+                    ]]
+                    last_save_time = self.database.get_last_balance_save_time(cursor)
+                    for chain in EVM_CHAIN_IDS_WITH_TRANSACTIONS:
+                        if len(location_labels := cursor.execute(  # number of earliest receive events of distinct assets after last_save_time  # noqa: E501
+                            'SELECT DISTINCT location_label FROM '
+                            '(SELECT MIN(timestamp), location_label FROM history_events '
+                            f'WHERE (type IN ({",".join(["?"] * len(type_filter))}) OR '
+                            f'subtype IN ({",".join(["?"] * len(subtype_filter))})) '
```

Comment:
One thing, the location label can't only be addresses. For kraken I believe is the exchange name. YOu need to add additional filters to limit the location to `EVM_LOCATIONS_TYPE`

Reply:
we filter the location using `Location.from_chain_id(chain).serialize_for_db()` where chain is from `EVM_CHAIN_IDS_WITH_TRANSACTIONS`, so we should only get the evm events